### PR TITLE
NativeAOT-LLVM: Support nativelib for Wasm 

### PR DESCRIPTION
**Pull request by yowl**
This PR removes the restriction of having a main enabling the `--nativelib` option.
Fixes a bug with a nativelib where the shadow stack would grow for every pinvoke call from the nativelib.
Fixes a bug processing options to enable `--codegenopt:Target=wasm32-unknown-unknown`

cc @SingleAccretion.

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs
Comment (SingleAccretion) [inline src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs]: Should this be `_nativeLib` for consistency?

Comment (yowl) [reply src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs]: yes, changed.

**Comment by tjpalmer** [comment]
Is there an eta on merging this?

File: src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs]:
```suggestion
        private bool _nativeLib;
```

File: src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs]:
```suggestion
            _nativeLib = nativeLib;
```

File: src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs]:
```suggestion
            return new LLVMCodegenCompilation(graph, factory, _compilationRoots, _ilProvider, _debugInformationProvider, _logger, _config, _inliningPolicy, _devirtualizationManager, _instructionSetSupport, _nativeLib);
```

**Review by MichalStrehovsky — APPROVED** [review]

**Comment by yowl** [comment]
@MichalStrehovsky @jkotas 

Thanks for approving, would it be ok to merge it?

**Comment by MichalStrehovsky** [comment]
> Thanks for approving, would it be ok to merge it?

Yeah; couldn't merge right away because I made changes and had to wait for the CI.

**Comment by yowl** [comment]
Thankyou!

**Pull request merged**
